### ISSUE TITLE
Almost half of the screen is black when barcodescanner is running. IOS

### ISSUE THREAD

**Issue by Gusten** [phonegap/phonegap-plugin-barcodescanner#244]
I'm building an app towards Android and IOS with Cordova and Ionic.
I've been using this plugin which has been working fine on Android but now when I'm trying to build on an IOS there is an issue with almost half of the screen being completely black and the other half showing the camera feed. See figure below

![img_0168](https://cloud.githubusercontent.com/assets/1529107/15606403/4e231bec-240c-11e6-9827-fd1fe3332c40.jpg)

I'm using the following Cordova plugins:

```json
"cordovaPlugins": [
    "cordova-plugin-console",
    "cordova-plugin-device",
    "https://github.com/apache/cordova-plugin-camera.git",
    "cordova-plugin-splashscreen",
    "ionic-plugin-keyboard",
    "phonegap-plugin-barcodescanner",
    "cordova-plugin-crosswalk-webview@1.6.1",
    "cordova-plugin-whitelist"
  ]
```

Does anyone have a clue what is going on and how I can fix it?

EDIT: Whops, forgot to mention this was running on an iPhone 6 with IOS 9.3.2

**Comment by Riaz12** [comment]
Hi Gusten,

Just want to find out, how did you center the camera and add the green border ?
Is that a default setting on iOS ?

Thanks

**Comment by daserge** [comment]
Confirmed.
The issue occurs if the video preview is opened when the app in the landscape orientation, f.e. if the orientation is unlocked:
```xml
<platform name="ios">
    <preference name="Orientation" value="all" />
</platform>
```

**Comment by daserge** [comment]
@Gusten, ios does not currently support scanning in orientations other that portrait so you could use this workaround for now:
`cordova plugin add cordova-plugin-screen-orientation`

```javascript
        document.getElementById('scan').addEventListener('click', function () {
            screen.lockOrientation('portrait');
            cordova.plugins.barcodeScanner.scan(
              function (result) {
                  screen.unlockOrientation();
                  console.log("We got a barcode\n" +
                        "Result: " + result.text + "\n" +
                        "Format: " + result.format + "\n" +
                        "Cancelled: " + result.cancelled);
              },
              function (error) {
                  screen.unlockOrientation();
                  console.log("Scanning failed: " + error);
              }
           );
        });
```

**Comment by daserge** [comment]
@Gusten, I've actually tested the #171 - and it fixes this issue, works great without the need to lock the orientation.
It would be great if you could confirm it also works in your case:
`cordova plugin add https://github.com/irontec/phonegap-plugin-barcodescanner`

**Comment by EddyVerbruggen** [comment]
Fixed in 5.0.1.
